### DISCUSSION TITLE
How to augment `request.routeConfig` type?

### DISCUSSION THREAD

**Discussion by bestickley** [fastify/fastify#5552] [Q&A] [answered]
Hi, I'd like to augment the `request.routeConfig` (docs [here](https://fastify.dev/docs/latest/Reference/Routes/#config)) but when I do so with the below code:

```ts
declare module "fastify" {
  interface FastifyRequest {
    jwt: DefaultJWT;
    routeConfig: FastifyRequest["routeConfig"] & { permissions: string[] };
  }
}
```

I get:
<img width="915" alt="image" src="https://github.com/fastify/fastify/assets/35735118/6c94af3d-e449-4395-b7d4-f522414cd700">

How can I do this?

**Comment by mcollina** [comment]
Thanks for reporting! I have no idea tbh, I think we might want to allow custom keys there.

**Reply by metcoder95** [reply]
Hi @bestickley would you like to submit a PR/Issue for this one? I believe we can also externalize this into an interface that can be augmented (just fast thinking, we might need to double-check)

**Reply by bestickley** [reply]
@metcoder95, I'm not sure this is possible anymore. See [here](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#:~:text=If%20they%20are%20not%20unique,overload%20of%20the%20same%20function.).

**Reply by metcoder95** [reply]
But you are re-declaring the module, and the interfaces should be merged; have we already tested that?

**Comment by bestickley** [comment, marked as answer]
Got this to work! See below:
```
declare module "fastify" {
  interface FastifyRequest {
    jwt: JWT;
  }
  interface FastifyContextConfig {
    tsRestRoute?: {
      metadata?: {
        permissions?: string[];
      };
    };
  }
}
```
Principle: if you want to augment a type within a lib, you need to redeclare an **interface** of the nested type your trying to augment. You won't be able to easily augment the sub-property as shown in my first attempt.
